quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java

```diff
@@ -382,57 +382,57 @@ && isUserModifiable(Configuration.getConfigValue(key))) {
             }
             if (options.includeRuntime) {
                 disabledMappers.addAll(PropertyMappers.getDisabledRuntimeMappers().values());
-            } else {
-                checkRuntimeSpiOptions(options, ignoredRunTime);
             }
 
-            for (OptionCategory category : abstractCommand.getOptionCategories()) {
-                List<PropertyMapper<?>> mappers = new ArrayList<>(disabledMappers);
+            var categories = new HashSet<>(abstractCommand.getOptionCategories());
+
+            // first validate the advertised property names
+            // - this allows for efficient resolution of wildcard values and checking spi options
+            Configuration.getConfig().getPropertyNames().forEach(name -> {
```

Comment:
Why is it better for wildcard mappers to go over all the property names instead of going over the wildcard property mappers? We already know what property mapper is the "wildcard" mapper, right? 

`PropertyMappers.getWildcardMappers()`

Reply:
We are trying to traverse over the property names only once. When there is a more efficient routine for getting the mapper, this will have roughly the cost of a constant number of map lookups. If we instead go over each wildcard mapper as it is currently written, each of the wilcard mappers + the spi logic will need to iterate over the property names.

Reply:
Right, make sense.